feat(pending): revoke pending values (#225)
morpho-org/metamorpho: README.md

Only one address can have this role.

It can:

- Do what the curator can do.

> - Do what the guardian can do.

Reviewer: except revoke the guardian!

Author: Yes, the owner can revoke the guardian

Only one address can have this role.

It can:

- Do what allocators can do.
- [Timelocked] Enable or disable a market by setting a supply cap to a specific market.
  - The supply cap must be set to 0 to disable the market.
  - Disabling a market can then only be done if the vault has no liquidity supplied on the market.
- Revoke the pending cap of any market.

#### Allocator

Multiple addresses can have this role.

It can:

- Set the `supplyQueue` and `withdrawQueue`, i.e. decide on the order of the markets to supply/withdraw from.
  - Upon a deposit, the vault will supply up to the cap of each Morpho Blue market in the `supplyQueue`` in the order set. The remaining funds are left as idle supply on the vault (uncapped).
  - Upon a withdrawal, the vault will first withdraw from the idle supply and then withdraw up to the liquidity of each Morpho Blue market in the `withdrawalQueue` in the order set.
  - The `supplyQueue` only contains markets which cap has previously been non-zero.
  - The `withdrawQueue` contains all markets that have a non-zero cap or a non-zero vault allocation.

> - Instantaneously reallocate funds across markets of the `withdrawQueue` at any moment.

Reviewer 2: [I wish this was the case.](https://github.com/morpho-org/metamorpho/issues/219)

Right now it is actually:
```suggestion
- Instantaneously reallocate funds by supplying on markets of the `withdrawQueue` and withdrawing on markets that have the same loan asset as the vault's asset.
```

Only one address can have this role.

It can:

- Do what allocators can do.
- [Timelocked] Enable or disable a market by setting a supply cap to a specific market.
  - The supply cap must be set to 0 to disable the market.
  - Disabling a market can then only be done if the vault has no liquidity supplied on the market.
- Revoke the pending cap of any market.

#### Allocator

Multiple addresses can have this role.

It can:

- Set the `supplyQueue` and `withdrawQueue`, i.e. decide on the order of the markets to supply/withdraw from.
  - Upon a deposit, the vault will supply up to the cap of each Morpho Blue market in the supply queue in the order set. The remaining funds are left as idle supply on the vault (uncapped).
  - Upon a withdrawal, the vault will first withdraw from the idle supply and then withdraw up to the liquidity of each Morpho Blue market in the withdrawal queue in the order set.

>   - The `supplyQueue` contains only enabled markets (enabled market are markets with non-zero cap or with non-zero vault's supply).

Reviewer 2: This is not true because a market supply cap can be set to non-zero, that market is added to the supply queue, then the supply cap can be put to zero again (before anyone has had the chance to supply on it) but it would not be removed from the supply queue. One invariant we have though is that markets from the supply queue have been previously approved (set with non-zero supply cap) by the curator/owner

Reviewer 2: Also the line "Instantaneously reallocate funds across enabled markets at any moment" is not accurate

Author: What's the problem with "Instantaneously reallocate funds across enabled markets at any moment"?

It can:

- Revoke the pending timelock.

> - Revoke the pending guardian (in particular, it can revoke any submission of another guardian).

Reviewer 2: ```suggestion
- Revoke the pending guardian (which means it can revoke any attempt to change the guardian).
```
(optional) I think it's clearer, even though it is not using the word `submit`
